charts/jaeger/templates/es-user-config.yaml

```diff
@@ -0,0 +1,62 @@
+{{ if (and (not .Values.userconfig) .Values.provisionDataStore.elasticsearch) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-config
+  namespace: {{ include "jaeger.namespace" . }}
+  labels:
+    {{- include "jaeger.labels" . | nindent 4 }}
+data:
+  user-config.yaml: |
+    service:
+      extensions: [jaeger_storage, jaeger_query, healthcheckv2]
+      pipelines:
+        traces:
+          receivers: [otlp]
+          processors: [batch]
+          exporters: [jaeger_storage_exporter]
+
+    extensions:
+      healthcheckv2:
+        use_v2: true
+        http:
+            endpoint: 0.0.0.0:13133
+
+      jaeger_query:
+        storage:
+          traces: some_storage
+          traces_archive: another_storage
+
+
+      jaeger_storage:
+        backends:
+          some_storage:
+            elasticsearch:
+              index_prefix: {{- .Values.storage.elasticsearch.index_prefix | quote | indent 2 }}
```

Comment:
I don't like this. `Values.storage.elasticsearch.index_prefix` is a separate domain-specific configuration language that exists only in the Operator - why do we need it? The users can provide their own custom config if they want to customize these things. If we still want to allow overriding specific fields of the built-in configuration, then those overrides should be using the OTEL config syntax, e.g. `Values.config.extensions.jaeger_storage...etc...`

Reply:
I wanted to keep the flag syntax same as jaeger-v1 so if the users wants to run it with old flag he can set it that way and helm will use `es-user-conifg.yaml `for that but if the user wants to provide there own config he can use the `-set-file` flag in that case we will use `user-config.yaml`